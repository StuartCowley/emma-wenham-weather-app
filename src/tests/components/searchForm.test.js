import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("Search Form", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchText=""
        setSearchText={setSearchText}
        onSubmit={handleCitySearch}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("button displays correct text", () => {
    const { getByText } = render(
      <SearchForm
        searchText=""
        setSearchText={setSearchText}
        onSubmit={handleCitySearch}
      />
    );

    expect(getByText("Search")).toBeInstanceOf(HTMLButtonElement);
  });
});
