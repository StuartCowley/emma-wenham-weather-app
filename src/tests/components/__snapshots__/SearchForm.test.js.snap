// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Summaries renders correctly`] = `
<DocumentFragment>
    <div
      class="search-form"
    >
      <label
        for="search"
        >
        <input
           placeholder="Search City"
           type="text"
           value=""
           />
        </label>
    <button
       type="submit"
         >
         Search
        </button> 
    </div>        
</DocumentFragment>
`;